src/site/content/en/blog/asyncify/index.md

```diff
@@ -0,0 +1,516 @@
+a file, and greet them with a "Hello, (username)!" message:
+
+```cpp
+#include <stdio.h>
+
+int main() {
+    char name[20+1];
+    FILE * stream;
+    stream = fopen("name.txt", "r");
+    size_t len = fread(&name, 1, 20, stream);
+    name[len] = '\0';
+    fclose(stream);
+    printf("Hello, %s!\n", name);
+    return 0;
+}
+```
+
+While it doesn't do much, it already demonstrates something you'll find in an application of any
+size: it reads some inputs from the external world, processes them internally and writes outputs
+back to the external world. All such interaction with the outside world happens via a few functions
+- commonly called input-output functions, also shortened to I/O.
+
+In order to read the name from C, you need at least two crucial I/O calls: `fopen`, to open the
+file, and `fread` to read the data from it. Once you retrieve the data, you can use another I/O
+function `printf` to print the result to the console.
+
+Those functions look quite simple at first glance and you don't have to think twice about all the
+machinery involved in order to read or write data. However, depending on the environment, there can
+be quite a lot going on inside:
+
+  - If the input file is located on a local drive, the application needs to perform a series of
+    memory and disk accesses to locate the file, check permissions, open it for reading, and then
+    read block by block until the requested number of bytes is retrieved. This can be pretty slow,
+    depending on the speed of your disk and the requested size.
+  - Or, the input file might be located on a mounted network location—in which case the network
+    stack will now be involved too, increasing the complexity, latency and number of potential
+    retries for each operation.
+  - Finally, even printf is not guaranteed to print things to the console and might be redirected to
+    a file or a network location, in which case it would have to go via the same steps above.
+
+Long story short, I/O can be slow and you can't predict how long a particular call will take by a
+quick glance at the code. While that operation is running, your whole application will appear frozen
+and unresponsive to the user.
+
+This is not limited to C or C++ either. Most system languages present all the I/O in a form of
+synchronous APIs. For example, if you translate the example to Rust, the API might look simpler, but
+the same principles apply. You just make a call and synchronously wait for it to return the result,
+while it performs all the expensive operations and eventually returns the result back to you in a
+single invocation:
+
+```rust
+fn main() {
+    let s = std::fs::read_to_string("name.txt");
+    println!("Hello, {}!", s);
+}
+```
+
+But what happens when you try to compile any of those samples to WebAssembly and translate them to
+the Web? Or, to provide a specific example, what could "file read" operation translate to? It would
```

Comment:
I think it's more of a "could" than "would", because there are many possibilities, and the one I'm describing is not the one currently used in toolchains.